qiskit/pulse/channels.py

```diff
@@ -153,8 +158,8 @@ def __eq__(self, other: 'Channel') -> bool:
         return type(self) is type(other) and self._index == other._index
 
     def __hash__(self):
-        if self._hash is None:
-            self._hash = hash((type(self), self._index))
+        # this is used to equality check of unordered channels of schedule
```

Comment:
It could also be used for dict keys, or anywhere else a hash is useful. Why specify one case?
```suggestion
        # this is used for equality checks of unordered schedule channels
```

Reply:
Because instruction hash should take subroutine into account, but this is heavy overhead. This is kind of lightweight hash to evaluate if parameter is updated.